udata/core/site/models.py

```diff
@@ -23,3 +28,30 @@ class Site(WithMetrics, db.Document):
     configs = db.DictField()
     themes = db.DictField()
     settings = db.EmbeddedDocumentField(SiteSettings, default=SiteSettings)
+
+
+def get_current_site():
+    if getattr(g, 'site', None) is None:
+        site_id = current_app.config['SITE_ID']
+        g.site, _ = Site.objects.get_or_create(id=site_id, defaults={
+            'title': current_app.config.get('SITE_TITLE'),
+            'keywords': current_app.config.get('SITE_KEYWORDS', []),
+        })
+    return g.site
+
+
+current_site = LocalProxy(get_current_site)
+
+
+@Dataset.on_delete.connect
+def pull_home_dataset(dataset):
```

Comment:
But it's not a removal from cache, this is simply the reference that is pulled from site model.
[Pull is a term used by Mongo for removing an item from a list](https://docs.mongodb.com/manual/reference/operator/update/pull/) but I can name it `remove_from_home_datasets` if you prefer.